src/caffeine/caffeine.c

```diff
@@ -53,110 +53,77 @@ void c_sync_all()
   gasnet_barrier_notify(0,GASNET_BARRIERFLAG_ANONYMOUS);
   gasnet_barrier_wait(0,GASNET_BARRIERFLAG_ANONYMOUS);
 }
-void c_co_sum_no_result_image_int32(void* c_loc_a, size_t Nelem, int* stat)
+
+void c_co_reduce_universal(void* c_loc_a, size_t Nelem, int* stat, int* result_image, int gex_dt, size_t sizeof_type, int gex_op)
```

Comment:
As a general rule I'd advise you not to define identifiers with a `gex_` or `gasnet_` name prefix. 

These name prefixes "belong" to GASNet, so it's not impossible that someday the GASNet headers might add a macro or function symbol aliasing the variable names you've chosen above. This could lead to compile errors in files including the GASNet headers, or even link-time symbol conflicts if both are extern symbols.

If you're looking for something that "says GASNet" but is not reserved, perhaps use `gn_`?

Reply:
Good suggestion.  I think I'll just drop the prefixes.

Reply:
@bonachea commit a1951b4 dropped the prefixes as suggested.